---
title: "Schober & Atanassov et al.: A quantitative map of mitochondrial protein phosphorylation upon loss of OXPHOS subunits"
output:
  html_document:
    code_folding: hide
    toc: TRUE
    
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	warning = FALSE,
	tidy = TRUE
)

# Packages used throughout the code
library("ggplot2")
library("ggrepel")
library("KEGGREST")
library("pathview")
library("UniProt.ws")
#load("UniProt.ws.Drosophila")
library("ggridges")
library("forcats")
library("ggpubr")
library("ggseqlogo")
library("stringr")
library("limma")
library("org.Dm.eg.db")
library("gplots")
library("pheatmap")
library("ggalluvial")

# The data analysis is based on several result files. Upon publication, the code and associated result files will be available as an R package.
# Upon request, we provide this R package also before, please e-mail florian.schober@ki.se

# Read functions:
source("SILAF_header.R")
```

# Figure 1 and S1: Labelling efficiency
The first section contains information about the general usability of the SILA fly in terms of labelling efficiency from egg to fly and within flies. 

## Figure 1A
Labelling efficiency from egg to fly - How fast is heavy lysine implemented in Drosophila larvae proteins? 

```{r FIGURE 1A, echo = TRUE, tidy = TRUE, warning = FALSE}
# Read basic data table and meta information
info <- read.table("info.txt", sep = "\t", header = T)
total <- read.table("proteinGroups.txt", sep = "\t", header = T)
rownames(total) <- total$Protein.IDs

# Remove contaminants
total <- total[!total$Potential.contaminant == "+",]
total <- total[!total$Reverse == "+",]
total <- total[!total$Only.identified.by.site == "+",]

# Get the intensity columns for SILAC values
colnames <- sub("Intensity.*\\.", "", colnames(total)) 

# Which files correspond to this experiment? Narrow down dataset
files <- info[info$Experiment=="QC.TIMESERIES",]$Tube 
timeseries <- total[,colnames %in% files]
rownames(timeseries) <- rownames(total)

# Making a data frame that is readable for density plots
for (i in 1:length(files)){
  if(i==1){
    histodata <- data.frame(1,1) # Data.frame for all density data at the beginning of the loop
    colnames(histodata) <- c("propheavy", "sample")
  }
  
  tot <- 3*i-2 # Specifies the columns to work in
  timeseries.current <- timeseries[,tot:(tot+2)] # Narrow down to a particular sample
  
  name <- substring(colnames(timeseries.current), 11)[1] # Extract sample name
  timeseries.current$sample <- name # Put the current sample name in each cell
  timeseries.current$propheavy <- as.numeric(as.character(timeseries.current[,3]/timeseries.current[,1])) # Calculate proportions
  timeseries.current <- timeseries.current[!is.na(timeseries.current$propheavy),] # Remove the NaN values that have not been picked up in that particular sample
  
  histodata.temp <- timeseries.current[,4:5] # Narrow down the number of columns for density plots
  assign(paste("timeseries.", name, sep=""), timeseries.current) # Give the variable a proper name
  
  histodata <- rbind(histodata, histodata.temp) # Add the the data to the old sample
  rownames(histodata) <- NULL
  
  # Modify the data.frame and add information about timepoints
  if(i == 14){
    histodata <- histodata[-1,] # Remove the first row after running the loop
    
    # And replace the samplenames with the correct identifieres:
    histodata$file <- histodata$sample
    
    for (i in files){
      age.temp <- as.character(info[as.character(info$Tube) %in% i,]$Condition)
      histodata$sample <- ifelse(histodata$file == i, age.temp, histodata$sample)
    }
    rm(histodata.temp, timeseries.current, age.temp) # Remove the intermediate files
  }
}

# Collecting the data summary per sample.
timeseries.mean <- data.frame(t(matrix(c(mean(timeseries.95$propheavy), mean(timeseries.102$propheavy), 
                                         sd(timeseries.95$propheavy), sd(timeseries.102$propheavy),
                                         mean(timeseries.96$propheavy), mean(timeseries.103$propheavy), 
                                         sd(timeseries.96$propheavy), sd(timeseries.103$propheavy),
                                         mean(timeseries.97$propheavy), mean(timeseries.104$propheavy),
                                         sd(timeseries.97$propheavy), sd(timeseries.104$propheavy),
                                         mean(timeseries.98$propheavy), mean(timeseries.105$propheavy),
                                         sd(timeseries.98$propheavy), sd(timeseries.105$propheavy),
                                         mean(timeseries.99$propheavy), mean(timeseries.106$propheavy),
                                         sd(timeseries.99$propheavy), sd(timeseries.106$propheavy),
                                         mean(timeseries.100$propheavy), mean(timeseries.107$propheavy),
                                         sd(timeseries.100$propheavy), sd(timeseries.107$propheavy),
                                         mean(timeseries.101$propheavy), mean(timeseries.108$propheavy),
                                         sd(timeseries.101$propheavy), sd(timeseries.108$propheavy)),
                                       nrow = 4, ncol = 7)))
timeseries.mean$day <- c("2", "4","6","8","aPupae","Zflies F", "Zflies M")
colnames(timeseries.mean) <- c( "Rep1", "Rep2", "SD1", "SD2", "day")
rownames(timeseries.mean) <- timeseries.mean$day

# Figure 1A:
#cairo_pdf("Figure1A.pdf")
ggplot()+
  geom_pointrange(data = timeseries.mean, 
                  mapping = aes(x = day, y = Rep1, ymin = Rep1-SD1, ymax = Rep1+SD1), 
                  size=0.5, color="black", fill="white", shape=1, 
                  position = position_jitter(w = 0, h = 0))+
  geom_pointrange(data = timeseries.mean, 
                  mapping = aes(x = day, y = Rep2, ymin = Rep2-SD2, ymax = Rep2+SD2), 
                  size=0.5, color="black", fill="white", shape=22, 
                  position = position_jitter(w = 0.4, h = 0))+
  coord_cartesian(ylim=c(0,1)) +
  scale_y_continuous(name="Labelling efficiency", breaks = c(0,.2,.4,.6,.8,1))+
  geom_hline(yintercept=0.95, lty = "dotted")+
  theme_bw()+
  ggtitle("Figure 1A: From egg to larvae labelling efficiency")
#dev.off()
```

## Figure 1B and 1C
Labelling efficiency on adult flies. 

```{r Figure 1B and 1C, message = FALSE, tidy = TRUE, warning = FALSE}
# Get the intensity columns for SILAC values
colnames <- sub("Intensity.*\\.", "", colnames(total)) 

# Which files correspond to this experiment?
files <- info[info$Experiment=="QC.TIMESERIES.FLIES",]$Tube 
timeseries <- total[,colnames %in% files]
rownames(timeseries) <- rownames(total)

### Making a data frame that is readable for density plots ###
for (i in 1:length(files)){
  if(i==1){
    histodata <- data.frame(1,1) # Data.frame for all density data at the beginning of the loop
    colnames(histodata) <- c("propheavy", "sample")
  }
  
  tot <- 3*i-2 # Specifies the columns to work in
  timeseries.current <- timeseries[,tot:(tot+2)] # Narrow down to particular sample
  
  name <- substring(colnames(timeseries.current), 11)[1] # Extract sample name
  timeseries.current$sample <- name # Put the current sample name in each cell
  timeseries.current$propheavy <- as.numeric(as.character(timeseries.current[,3]/timeseries.current[,1])) # Calculate proportions
  timeseries.current <- timeseries.current[!is.na(timeseries.current$propheavy),] # Remove the NaN values that have not been picked up in that particular sample
  
  histodata.temp <- timeseries.current[,4:5] # Narrow down the number of columns for density plots
  assign(paste("timeseries.", name, sep=""), timeseries.current) # Give the variable a proper name
  
  histodata <- rbind(histodata, histodata.temp) # Add the the data to the old sample
  rownames(histodata) <- NULL
  
  # Modify the data.frame and add information about timepoints
  if(i == 14){
    histodata <- histodata[-1,] # Remove the first row after running the loop
    
    # And replace the samplenames with the correct identifieres:
    histodata$file <- histodata$sample
    
    for (i in files){
      age.temp <- as.character(info[as.character(info$Tube) %in% i,]$Condition)
      histodata$sample <- ifelse(histodata$file == i, age.temp, histodata$sample)
    }
    rm(histodata.temp, timeseries.current, age.temp) # Remove the intermediate files
  }
}

ggplot(data = timeseries.152, aes(x = propheavy, y = log2(Intensity.152)))+
  geom_point()+
  geom_point(data = timeseries.152[rownames(timeseries.152) == "Q9VPE2",],color = "red")

### Collecting the data summary per sample. Not the most elegant chunk of code ever written. ###
timeseries.mean <- data.frame(t(matrix(c(mean(timeseries.146$propheavy), mean(timeseries.153$propheavy), 
                                         sd(timeseries.146$propheavy),  sd(timeseries.153$propheavy),
                                         mean(timeseries.147$propheavy),  mean(timeseries.154$propheavy), 
                                         sd(timeseries.147$propheavy),  sd(timeseries.154$propheavy),
                                         mean(timeseries.148$propheavy), mean(timeseries.155$propheavy),
                                         sd(timeseries.148$propheavy), sd(timeseries.155$propheavy),
                                         mean(timeseries.149$propheavy), mean(timeseries.156$propheavy),
                                         sd(timeseries.149$propheavy), sd(timeseries.156$propheavy),
                                         mean(timeseries.150$propheavy), mean(timeseries.157$propheavy),
                                         sd(timeseries.150$propheavy), sd(timeseries.157$propheavy),
                                         mean(timeseries.151$propheavy), mean(timeseries.158$propheavy),
                                         sd(timeseries.151$propheavy), sd(timeseries.158$propheavy),
                                         mean(timeseries.152$propheavy), mean(timeseries.159$propheavy),
                                         sd(timeseries.152$propheavy),sd(timeseries.159$propheavy)), nrow = 4, ncol = 7)))
timeseries.mean$day <- c("d02", "d04", "d06","d08","d10","d12", "d14")
colnames(timeseries.mean) <- c( "Rep1", "Rep2", "SD1", "SD2", "day")
rownames(timeseries.mean) <- timeseries.mean$day

# Figure 1B
#pdf("Figure1B.pdf")
ggplot(histodata, aes(x = propheavy, y = fct_rev(sample))) + 
  geom_density_ridges_gradient(aes(fill = ..x..), scale = 3, size = 0.5, panel_scaling = T)+
  scale_fill_gradientn(colours = c("white", "#F2A7A7", "#D54036"), name = "Labelling efficiency")+
  theme_ridges(font_size = 13, grid = T) + theme(axis.title.y = element_blank())+
  xlab("Labelling efficiency")+
  scale_x_continuous(breaks=seq(from = 0, to = 1, by = 0.2))+
  ggtitle("Figure 1B: Timecourse labelling efficiency in flies")+
  geom_vline(xintercept = apply(timeseries.mean[,1:2], 1, mean))
#dev.off()

# Additional figure
# ggplot()+
#   geom_pointrange(data = timeseries.mean, 
#                   mapping = aes(x = day, y = Rep1, ymin = Rep1-SD1, ymax = Rep1+SD1), 
#                   size=0.5, color="black", fill="white", shape=1, 
#                   position = position_jitter(w = 0, h = 0))+
#   geom_pointrange(data = timeseries.mean, 
#                   mapping = aes(x = day, y = Rep2, ymin = Rep2-SD2, ymax = Rep2+SD2), 
#                   size=0.5, color="black", fill="white", shape=22, 
#                   position = position_jitter(w = 0.4, h = 0))+
#   coord_cartesian(ylim=c(0,1)) +
#   scale_y_continuous(name="Labelling efficiency", breaks = c(0,.2,.4,.6,.8,1))+
#   theme_classic()


# Figure 1C

histodata.limitedd2 <- histodata[histodata$sample=="d14",]
#pdf("Figure1C_part1.pdf")
ggplot(histodata.limitedd2, aes(x = propheavy, y = fct_rev(sample))) +
  geom_density_ridges_gradient(aes(fill = ..x..), scale = 3, size = 0.5, panel_scaling = T) +
  scale_fill_gradientn(colours = c("white"), name = "Labelling efficiency")+
  theme_ridges(font_size = 13, grid = T) + theme(axis.title.y = element_blank())+
  xlab("Labelling efficiency")+
  ggtitle("Figure 1C: Labelling efficiency in flies at 14d")+
  scale_x_continuous(breaks=seq(from = 0, to = 1, by = 0.2))
#dev.off()

# Which proteins get enriched in particular fractions?
fraction.identifier(timeseries.152, "152", timeseries.159, "159", "GSEA/Timeseries.flies", 1, 0.9)
fraction.identifier(timeseries.152, "152", timeseries.159, "159", "GSEA/Timeseries.flies", 0.9, 0.3)
fraction.identifier(timeseries.152, "152", timeseries.159, "159", "GSEA/Timeseries.flies", 0.3, 0.1)
fraction.identifier(timeseries.152, "152", timeseries.159, "159", "GSEA/Timeseries.flies", 0.1, 0)

# Plot the GSEA results
plot.fly.Ribosome <-  kegg.mapping(data = timeseries.152, suffix = "Ribosome", pathway = "03010")
plot.fly.OXPHOS <-  kegg.mapping(data = timeseries.152, "OXPHOS", "00190")
plot.fly.TCA <-  kegg.mapping(data = timeseries.152, "TCA", "00020")


# Has to be done seperately for the AA pathway as KEGG does not return members
AA <- c("42783","46391","35728","32940","40812","36020","32292","32297","43447","32087","33081","33351","46717","43582","48552","33172","37804","31524","246599","35692","33461","39085","31579","36060","33373","42689","43102","37143","42106","42284","43183","34554","41269","42586","42728","39132","42620","44291","38871","40443","41027","41117","36782","32545","44149","3771738","40097","39878")

# Metab has to be done seperately due to the structure of the KEGG pathway
data.entrez <- select(up, sub(";.*", "", rownames(timeseries.152)), "ENTREZ_GENE")
timeseries.152$entrez <- data.entrez[match(sub(";.*", "", rownames(timeseries.152)), data.entrez$UNIPROTKB),]$ENTREZ_GENE
interest.tmp <- timeseries.152[timeseries.152$entrez %in% AA,]

metab <- c("19893538","19893540","19893556","38753","47173","45880","42783","38723","37466","39402","38697","45786","38647","35146","44001","35153","39434","39424","38447","35279","38598","39470","38463","40679","37130","31695","33918","42973","32334","45398","35671","43829","35761","40492","40415","39479","40599","43713","33179","40875","32471","39899","35829","36098","40263","39213","43437","46391","43072","41067","38076","35728","38378","31805","40692","40630","32936","32940","38864","34613","41570","40708","33528","45668","32974","34587","36640","40812","34925","34995","36524","34159","43373","46040","40675","33117","32989","44007","31703","49953","36991","43671","41432","41468","34598","45577","38484","37166","37197","36020","50178","32292","32297","38315","35292","41173","36540","39740","43442","39396","43447","33510","37228","42466","37728","44183","35138","34997","32013","35507","41550","3355111","32087","3885565","35140","33351","3355069","37131","36932","30995","53511","3771779","46717","40528","35986","48224","43913","47895","41894","42216","31697","31817","43798","40836","43757","43750","43582","31858","43739","31960","53581","31762","48552","33172","40946","37804","40936","31342","31950","45875","246458","31524","246599","35692","33493","43244","42836","33461","326133","37834","40925","44307","33986","35265","35118","31604","326187","326188","326189","33443","317928","326209","40406","261603","318053","32822","32352","59261","35590","43191","37539","326264","37867","41745","2768916","3772179","317974","32582","44701","37699","5740593","39066","31578","3771877","33524","117361","32585","42047","32586","42051","31289","37617","31376","31190","37946","33626","31185","34256","45012","31579","40059","31587","44207","40056","32228","41253","3355155","36060","19988923","8673970","41279","39856","31605","31606","33642","39841","5740185","39846","33373","37044","36396","39403","41845","44154","39065","34276","42759","37419","42756","36844","44117","37784","34313","41313","39050","43092","37931","42364","41326","39820","192507","36847","39819","38612","34552","36955","32672","248194","37028","43102","41869","40012","44010","34016","34017","41360","34021","42832","37143","34414","39761","42834","43189","43350","34716","32434","39986","39985","32441","40241","32740","42505","42939","33742","42106","43936","42849","39976","42282","42284","36496","40272","42291","41896","43183","39304","43165","46069","39281","42341","42326","34554","43240","41269","42591","37029","38986","34474","42901","42586","32839","39214","36536","37517","41311","53507","38979","41333","41340","39160","42728","39132","32775","41823","39392","41433","33783","31406","42620","32789","34064","40434","34060","44291","40390","33386","32880","32887","44390","38871","39988","40188","37217","40443","39950","43507","40389","41605","36159","43005","40349","43516","34137","43950","38221","42185","35944","41027","36826","35923","35918","44008","41117","44702","39143","41149","36731","36760","41167","40167","41192","36782","36587","33744","35830","35825","35824","35790","46068","45401","36392","44228","33718","32545","36927","46059","31849","44046","33839","33847","33852","38147","38154","32565","33935","45830","44149","35384","35383","48450","34747","37385","45368","40348","40346","35586","37415","37435","32420","3771738","33898","33911","40097","39878","37445","41022","46260","33431","32592","38342","19893558","19893542","19893549")

interest.tmp <- timeseries.152[timeseries.152$entrez %in% metab,]
  
plot.fly.Metabolism <- ggplot(data = timeseries.152, aes (x = propheavy, y = log10(Intensity.152))) +
          geom_point(alpha = 0.2)+
          geom_point(data = interest.tmp, aes (x = propheavy, y = log10(Intensity.152)), color = "firebrick3")+
          geom_vline(xintercept = mean.na(interest.tmp$propheavy), color = "firebrick3")+
          theme_classic()+
          labs(title = "Metabolism", x = "Labelling efficiency", "log10(Intensity)")


# Figure S2C
#cairo_pdf("Figure1C_part2.pdf", width = 12, height = 3)
ggarrange(plot.fly.Metabolism, plot.fly.OXPHOS, plot.fly.Ribosome, plot.fly.TCA, nrow = 1, ncol = 4)
#dev.off()
```

## Figure 1D
What proteins at 2dae don't have a 100% incorporation efficiency? 

```{r Figure 1B , echo = T, tidy = TRUE, warning = FALSE}
# Limit dendrogram to the interesting candidate d2
histodata.limitedd2 <- histodata[histodata$sample=="d2",]

# Figure 1D
#pdf("Figure1D_part1.pdf")
ggplot(histodata.limitedd2, aes(x = propheavy, y = fct_rev(sample))) +
  geom_density_ridges_gradient(aes(fill = ..x..), scale = 3, size = 0.5, panel_scaling = T) +
  scale_fill_gradientn(colours = c("white"), name = "Labelling efficiency")+
  theme_ridges(font_size = 13, grid = T) + theme(axis.title.y = element_blank())+
  xlab("Labelling efficiency")+
  scale_x_continuous(breaks=seq(from = 0, to = 1, by = 0.2))
#dev.off()

# Identify proteins in relevant categories
#fraction.identifier(timeseries.95, "95", timeseries.102, "102", "GSEA/LOWTURNOVER", 0.9, 0.6)
#fraction.identifier(timeseries.95, "95", timeseries.102, "102", "GSEA/LOWTURNOVER", 0.6, 0.1)
#fraction.identifier(timeseries.95, "95", timeseries.102, "102", "GSEA/LOWTURNOVER", 0.1, 0)

plot.larvae.Ribosome <-  kegg.mapping(data = timeseries.95, "Ribosome", "03010")
plot.larvae.Proteasome <-  kegg.mapping(data = timeseries.95, "Proteasome", "03050")
plot.larvae.OXPHOS <- kegg.mapping(data = timeseries.95, "OXPHOS", "00190")
plot.larvae.Phagosome <- kegg.mapping(data = timeseries.95, "Phagosome", "04145")
plot.larvae.MAPK <- kegg.mapping(data = timeseries.95, "MAPK", "04013")
plot.larvae.ER <- kegg.mapping(data = timeseries.95, "ER", "04141")
plot.larvae.TCA <- kegg.mapping(data = timeseries.95, "TCA", "00020")

data.entrez <- select(up, sub(";.*", "", rownames(timeseries.95)), "ENTREZ_GENE")
timeseries.95$entrez <- data.entrez[match(sub(";.*", "", rownames(timeseries.95)), data.entrez$UNIPROTKB),]$ENTREZ_GENE
interest.tmp <- timeseries.95[timeseries.95$entrez %in% AA,]
  
plot.larvae.AA <- ggplot(data = timeseries.95, aes (x = propheavy, y = log10(Intensity.95))) +
          geom_point(alpha = 0.2)+
          geom_point(data = interest.tmp, aes (x = propheavy, y = log10(Intensity.95)), color = "firebrick3")+
          geom_vline(xintercept = mean.na(interest.tmp$propheavy), color = "firebrick3")+
          theme_classic()+
          labs(title = "AA", x = "Labelling efficiency", "log10(Intensity)")

#Figure 1D continued
#cairo_pdf("Figure1D_part2.pdf", width = 5.5, height = 3)
ggarrange(plot.larvae.Ribosome, plot.larvae.Proteasome, nrow = 1, ncol = 2)
#dev.off()

#cairo_pdf("Figure1D_part3.pdf", width = 5.5, height = 12)
  ggarrange(plot.larvae.AA, plot.larvae.TCA, plot.larvae.OXPHOS,
            plot.larvae.MAPK, plot.larvae.ER,plot.larvae.Phagosome, 
            nrow = 4, ncol = 2)
#dev.off()

```


# Figure S1: Quality control
(1) Why do we not use Arg/Lys double labelling? (2) Is SILAF as accurate as LFQ? The "read.ms" function is described in detail in the header file ("SILAF_header.R").

## Figure S1A -S1D
Conversion of heavy amino acids - The reason why we excluded Arg and focus on Lys6 only.

```{r Figure S1A to S1D, tidy = TRUE, warning = FALSE}

# Show conversion into Proline
library(tidyverse)

mod.spec.pept <- read_tsv("Proline/modificationSpecificPeptides.txt")
mod.spec.pept$containsK <- grepl("K", mod.spec.pept$Sequence)
mod.spec.pept$containsKR <- grepl("R|K", mod.spec.pept$Sequence)
mod.spec.pept$containsP <- grepl("P", mod.spec.pept$Sequence)

## Figure S1C1:
mod.spec.pept %>%
  filter(containsKR == TRUE) %>%
  mutate(`Modification state` = ifelse(Modifications %in% "Unmodified","Unmodified",
                                       ifelse(grepl("Pro6",Modifications),"Pro6 Modified","Arg / Lys Modified"))) %>% 
  group_by(`Raw file`,`Modification state`) %>% 
  summarise(Count = n()) %>% 
  ungroup() %>% 
  group_by(`Raw file`) %>% 
  mutate(`Normalized count [%]` = 100*(`Count`/sum(`Count`))) %>% 
  filter(grepl("31|37|101|108",`Raw file`)) %>% 
  ggplot(aes(x = `Raw file`, y = `Normalized count [%]`, fill = `Modification state`))+
  geom_bar(stat = "identity", color = "black")+
  geom_text(aes(label = round(`Normalized count [%]`,2)),position = position_stack(vjust = 0.5))+
  scale_fill_brewer(palette = "Paired")+
  theme_bw()+
  ggsave(filename = "Proline/ModificationstateKR.pdf")

## Figure S1C2:
mod.spec.pept %>%
  filter(containsK == TRUE) %>%
  mutate(`Modification state` = ifelse(Modifications %in% "Unmodified","Unmodified",
                                       ifelse(grepl("Pro6",Modifications),"Pro6 Modified","Arg / Lys Modified"))) %>% 
  group_by(`Raw file`,`Modification state`) %>% 
  summarise(Count = n()) %>% 
  ungroup() %>% 
  group_by(`Raw file`) %>% 
  mutate(`Normalized count [%]` = 100*(`Count`/sum(`Count`))) %>% 
  filter(grepl("31|37|101|108",`Raw file`)) %>% 
  ggplot(aes(x = `Raw file`, y = `Normalized count [%]`, fill = `Modification state`))+
  geom_bar(stat = "identity", color = "black")+
  geom_text(aes(label = round(`Normalized count [%]`,2)),position = position_stack(vjust = 0.5))+
  scale_fill_brewer(palette = "Paired")+
  theme_bw()+
  ggsave(filename = "Proline/ModificationstateK.pdf")

## Figure S1D1:
mod.spec.pept %>%
  filter(containsP == TRUE) %>%
  filter(containsK == TRUE) %>%
  mutate(`Modification state` = ifelse(Modifications %in% "Unmodified","Unmodified",
                                       ifelse(grepl("Pro6",Modifications),"Pro6 Modified","Arg / Lys Modified"))) %>% 
  group_by(`Raw file`,`Modification state`) %>% 
  summarise(Count = n()) %>% 
  ungroup() %>% 
  group_by(`Raw file`) %>% 
  mutate(`Normalized count [%]` = 100*(`Count`/sum(`Count`))) %>% 
  filter(grepl("31|37|101|108",`Raw file`)) %>% 
  ggplot(aes(x = `Raw file`, y = `Normalized count [%]`, fill = `Modification state`))+
  geom_bar(stat = "identity", color = "black")+
  geom_text(aes(label = round(`Normalized count [%]`,2)),position = position_stack(vjust = 0.5))+
  scale_fill_brewer(palette = "Paired")+
  theme_classic()+
  ggsave(filename = "Proline/ModificationstateP+K.pdf")

## Figure S1D2:
mod.spec.pept %>%
  filter(containsP == TRUE) %>%
  filter(containsKR == TRUE) %>%
  mutate(`Modification state` = ifelse(Modifications %in% "Unmodified","Unmodified",
                                       ifelse(grepl("Pro6",Modifications),"Pro6 Modified","Arg / Lys Modified"))) %>% 
  group_by(`Raw file`,`Modification state`) %>% 
  summarise(Count = n()) %>% 
  ungroup() %>% 
  group_by(`Raw file`) %>% 
  mutate(`Normalized count [%]` = 100*(`Count`/sum(`Count`))) %>% 
  filter(grepl("31|37|101|108",`Raw file`)) %>% 
  ggplot(aes(x = `Raw file`, y = `Normalized count [%]`, fill = `Modification state`))+
  geom_bar(stat = "identity", color = "black")+
  geom_text(aes(label = round(`Normalized count [%]`,2)),position = position_stack(vjust = 0.5))+
  scale_fill_brewer(palette = "Paired")+
  theme_classic()+
  ggsave(filename = "Proline/ModificationstateP+KR.pdf")

# Split violin plots part 1. Code taken from https://stackoverflow.com/questions/35717353/split-violin-plot-with-ggplot2 of user jan-glx
GeomSplitViolin <- ggproto("GeomSplitViolin", GeomViolin, draw_group = function(self, data, ..., draw_quantiles = NULL){
  data <- transform(data, xminv = x - violinwidth * (x - xmin), xmaxv = x + violinwidth * (xmax - x))
  grp <- data[1,'group']
  newdata <- plyr::arrange(transform(data, x = if(grp%%2==1) xminv else xmaxv), if(grp%%2==1) y else -y)
  newdata <- rbind(newdata[1, ], newdata, newdata[nrow(newdata), ], newdata[1, ])
  newdata[c(1,nrow(newdata)-1,nrow(newdata)), 'x'] <- round(newdata[1, 'x']) 
  if (length(draw_quantiles) > 0 & !scales::zero_range(range(data$y))) {
    stopifnot(all(draw_quantiles >= 0), all(draw_quantiles <= 
                                              1))
    quantiles <- ggplot2:::create_quantile_segment_frame(data, draw_quantiles)
    aesthetics <- data[rep(1, nrow(quantiles)), setdiff(names(data), c("x", "y")), drop = FALSE]
    aesthetics$alpha <- rep(1, nrow(quantiles))
    both <- cbind(quantiles, aesthetics)
    quantile_grob <- GeomPath$draw_panel(both, ...)
    ggplot2:::ggname("geom_split_violin", grid::grobTree(GeomPolygon$draw_panel(newdata, ...), quantile_grob))
  }
  else {
    ggplot2:::ggname("geom_split_violin", GeomPolygon$draw_panel(newdata, ...))
  }
})

# Split violin plots part 2. Code taken from https://stackoverflow.com/questions/35717353/split-violin-plot-with-ggplot2 of user jan-glx
geom_split_violin <- function (mapping = NULL, data = NULL, stat = "ydensity", position = "identity", ..., draw_quantiles = NULL, trim = TRUE, scale = "area", na.rm = FALSE, show.legend = NA, inherit.aes = TRUE) {
  layer(data = data, mapping = mapping, stat = stat, geom = GeomSplitViolin, position = position, show.legend = show.legend, inherit.aes = inherit.aes, params = list(trim = trim, scale = scale, draw_quantiles = draw_quantiles, na.rm = na.rm, ...))
}

# Look at non-normalized values in the six Arg/Lys samples and convert them to a different data.frame format readable for split violin
length <- length(total$Ratio.H.L.121)
argconv <- data.frame(x =   rep("Precision", length*2),
                      y =   c(total$Ratio.H.L.124,# total$Ratio.H.L.119, total$Ratio.H.L.120, 
                            total$Ratio.H.L.142),#, total$Ratio.H.L.143, total$Ratio.H.L.144),
                      fill= c(rep("Lys6", 1*length), rep("Arg10Lys6", 1*length)))

argconv$x <- as.factor(argconv$x)
argconv$y <- log2(argconv$y)

# Figure S1A
#pdf("FigureS1A.pdf", 5, 3)
ggplot(argconv, aes(x, y, fill=fill)) + 
  geom_hline(yintercept = 0) +
  geom_split_violin()+
  theme_classic()
#dev.off()

# Look at non-normalized values in six comparable Lys samples and convert them to a different data.frame format readable for split violin
lysCeffect <- data.frame(x = c(rep("Precision", length*6), rep("Monster", length*6)),
                      y =   c(total$Intensity.L.122, total$Intensity.L.123, total$Intensity.L.124, 
                            total$Intensity.L.142, total$Intensity.L.143, total$Intensity.L.144,
                            total$Intensity.L.118, total$Intensity.L.119, total$Intensity.L.120, 
                            total$Intensity.L.139, total$Intensity.L.140, total$Intensity.L.141),
                      fill= c(rep("Lys6", 3*length), rep("Arg10Lys6", 3*length),
                            rep("Lys6", 3*length), rep("Arg10Lys6", 3*length)))

# Figure S1B
#pdf("FigureS1B.pdf", 5, 3)
ggplot(lysCeffect, aes(x, log2(y), fill=fill)) + 
  geom_hline(yintercept = 0) +
  geom_split_violin()+
  theme_classic()
#dev.off()

detach("package:tidyverse", unload=TRUE)
```

## Figure S1E: Compare accuracy of SILAF with LFQ of the respective light fraction
```{r Figure S1E, tidy = TRUE, warning = FALSE}
## Calculate the SDs per protein measurement and combine in another dataframe
### Read in LFQ of SILAF.L and SILAF data
files.current <- info[info$Experiment=="QC.LFQ.SILAFL",][,"Tube"]
colnames <- sub("LFQ\\.intensity\\.L\\.", "", colnames(total)) #<Obtain columns that contain the quantification data.>
dataset.holidic.L <- total[,colnames %in% files.current]

files.current <- info[info$Experiment=="QC.SILAF.PRECISION",][,"Tube"]
colnames <- sub("Ratio\\.H\\.L\\.normalized\\.", "", colnames(total)) #<Obtain columns that contain the quantification data.>
dataset.silaf <- total[,colnames %in% files.current]

precision.LFQ.SILAF <- data.frame(dataset.holidic.L, dataset.silaf)
precision.LFQ.SILAF[precision.LFQ.SILAF == 0 | precision.LFQ.SILAF == "NaN"] <- NA

### Exclude incomplete columns
precision.LFQ.SILAF <- precision.LFQ.SILAF[complete.cases(precision.LFQ.SILAF),]
  
### Normalize the LFQ values to 1
precision.holidicL.mean <- apply(precision.LFQ.SILAF[,1:4], 1, mean)

df1 <- data.frame(precision.holidicL.sd, "LFQ.L")
rownames(df1) <- NULL
df2 <- data.frame(precision.SILAF.sd, "SILAF")
rownames(df2) <- NULL
df3 <- data.frame(precision.holidicH.sd, "LFQ.H")
rownames(df3) <- NULL

names(df1) <- names(df2)
precision.LFQ.SILAF.sd <- rbind(df1, df2)
colnames(precision.LFQ.SILAF.sd) <- c("SD", "dataset")

### Figure S1E: Plot the SD
#cairo_pdf("FigureS1E.pdf")
ggplot(precision.LFQ.SILAF.sd, aes(x = SD, fill = dataset)) +
  geom_density(alpha = 0.5)+
  coord_cartesian(xlim = c(0,1.0))+
  theme_classic()+
  scale_fill_manual(values = c("grey", "firebrick3"))
#dev.off()
```

## Parts of figure S1F/SILAF
Compare the intensity of light and heavy fraction in SILAF samples. Note that this code uses normalized H/L ratios to make it comparable to LFQ following below (which is intrinsically normalized.)

```{r Figure S1F top, message = FALSE, tidy = TRUE, warning = FALSE}
### (1) Read in data >>>
precision.SILAF.stats <- read.table("Preseus_stats/QC.SILAF.PRECISION.STATS.txt", header = T, sep = "\t")
precision.SILAF <- read.ms(data = total, meta = info, name = "QC.SILAF.PRECISION", type = "SILAF", sum.columns = "TRUE",
                           inf.clearing = "to.NA", plots = "TRUE", stats = precision.SILAF.stats, stat.test = "X.Log.t.test.p.value")

### (2) Split the samples into elements of a list. Necessary to generate seperate scatterplots >>>

file.counter <- length(unique(sub(".*\\.","",colnames(precision.SILAF[["int"]])))) # How many samples?

  ## Iterate column groups for one sample into one element of a list each. There are file.counter = i samples >>
  for (i in c(1:file.counter)){
    if(i==1){
      precision.SILAF.samples <- list() # Empty list in first round
    }
  data.current <- precision.SILAF[["int"]][,(i*3-2):(i*3)] # Narrow down to the current dataset
  data.current <- data.current[data.current[,1]!=0,] # Exclude all rows with a total count of 0
  sample.current <- sub(".*\\.","",colnames(data.current)[1]) # What's the samplename?
  precision.SILAF.samples[[sample.current]] <- data.current # Give the listelement the name of the current sample
  rm(data.current, sample.current)
  }

### (3) Generates one plot per element in the list >>>

#cairo_pdf("FigureS1F.pdf", 6,9)
par(mfrow = c(4,3))
for (i in c(1:file.counter)){
  panel.smooth(precision.SILAF.samples[[i]][,2], precision.SILAF.samples[[i]][,3])
  panel.hist(precision.SILAF.samples[[i]][,2])
  panel.hist(precision.SILAF.samples[[i]][,3])
}
#dev.off()

```

## Parts of figure S1G/LFQ on holdic flies
Flies that are grown on light, artificial food to obtain their correlation coefficients.

```{r Figure S1G bottom holidic, tidy = TRUE, warning = FALSE}
precision.LFQ.holidic <- read.ms(data = total, meta = info, name = "QC.LFQ.SILAFL", type = "LFQ", inf = "to.NA")
precision.LFQ.holidic <- precision.LFQ.holidic[,-c(2,4,6,8)]

#pdf("FigureS1G.pdf")
pairs(precision.LFQ.holidic,
      lower.panel = panel.smooth.add,
      diag.panel = panel.hist.add,
      upper.panel = NULL,
      main = "S1D: LFQ samples on holidic food")
#dev.off()
```

# Figure S2: Comparison of the results to previously published data by comparing male and female flies

## Fly/Larvae dynamic range, Figure S2A
```{r Dynamic range}
dynamic <- timeseries[,!grepl(".*[LH].*", colnames(timeseries))]

library("tidyverse")
dynamic <- gather(dynamic, "sample", "Intensity")
dynamic <- arrange(dynamic, sample, Intensity)
dynamic[dynamic == 0] <- NA
dynamic <- dynamic[!is.na(dynamic$Intensity),]

dynamic <- dynamic[as.numeric(sub("Intensity\\.", "", dynamic$sample)) > 101,] # Include just one replicate for better understanding the data

a <- c()
for(i in c(1:7)){
  v <- c(1:as.vector(table(dynamic$sample))[i])
  a <- c(a,v)
}

dynamic$order <- a

#cairo_pdf("FigureS2A.pdf", 5, 5)
ggplot(data = dynamic, aes (x = order, y = log2(Intensity), col = sample))+
  geom_point()+
  scale_color_brewer(palette = "Reds")+
  theme_bw()
#dev.off()
detach("package:tidyverse", unload=TRUE)

```

## Sex specific proteome: Figure S2B-D
... which has been done in the original SILAC fly paper from 2010 (Sury et al.). We try to reprocude the results and align them to the old, re-analysed data.

```{r Figure S2, tidy = TRUE, warning = FALSE}
# Read data. The dataset was collected seperately and is thus not included in the total dataset used above
frac.flysex <- read.table("MaleFemale/Frac/with_requantify/proteinGroups.txt", header = T, sep = "\t")
flysex.stats <- data.frame(Ratio.H.L.normalized.0913_QE_IA_NL_FS_125 = frac.flysex$Ratio.H.L.normalized.SILAF.F.vs.M.Fod,
                           Ratio.H.L.normalized.0913_QE_IA_NL_FS_126 = frac.flysex$Ratio.H.L.normalized.SILAF.F.vs.M.Rev,
                           Protein.IDs = frac.flysex$Protein.IDs,
                           Gene.names = frac.flysex$Gene.names,
                           protein.names = frac.flysex$Protein.names,
                           Rev = frac.flysex$Reverse,
                           Contm = frac.flysex$Potential.contaminant,
                           Site = frac.flysex$Only.identified.by.site,
                           Int = apply(frac.flysex[,c(88,89,91,92)], 1, mean.na))

#Exclusion criteria
flysex.stats <- flysex.stats[(flysex.stats$Rev != "+"),]
flysex.stats <- flysex.stats[(flysex.stats$Contm != "+"),]
flysex.stats <- flysex.stats[(flysex.stats$Site != "+"),]

flysex.stats[,1:2] <- log2(flysex.stats[,1:2])
flysex.stats$Ratio.H.L.normalized.0913_QE_IA_NL_FS_126 <- -flysex.stats$Ratio.H.L.normalized.0913_QE_IA_NL_FS_126

# Entrez ID mapping
entrez <- select(up, sub(";.*", "", flysex.stats$Protein.IDs), "ENTREZ_GENE")
entrez <- entrez[!duplicated(entrez$UNIPROTKB),]
flysex.stats$entrez <- entrez$ENTREZ_GENE[match(entrez$UNIPROTKB, sub(";.*", "", flysex.stats$Protein.IDs))]

# Statistics
fit <- lmFit(flysex.stats[,1:2])
fit <- eBayes(fit)
limma.result <- topTable(fit,number = Inf,confint = TRUE, adjust.method = "fdr")

flysex.stats$FDR <- limma.result[match(rownames(flysex.stats), rownames(limma.result)),]$adj.P.Val
flysex.stats$mean <- apply(flysex.stats[,1:2], 1, mean.na)

#write.table(flysex.stats, "/Users/florsc/Documents/1_Projects/_SAMC/SILAF/Submission/Flysex.txt",  sep = "\t")

# How are the two replicates towards each other: Figure S3A
select = abs(apply(flysex.stats[,1:2], 1, mean.na)) > 4.0 # The ones to highlight

#Calculate R:
round(cor(flysex.stats$Ratio.H.L.normalized.0913_QE_IA_NL_FS_125, flysex.stats$Ratio.H.L.normalized.0913_QE_IA_NL_FS_126, use = "complete.obs"), digits=4)

#cairo_pdf("FigureS2B.pdf")
ggplot(data = flysex.stats, aes(x = Ratio.H.L.normalized.0913_QE_IA_NL_FS_125, y = Ratio.H.L.normalized.0913_QE_IA_NL_FS_126))+
  geom_abline(intercept = 0, slope = 1, lty = "dotted")+
  geom_hline(yintercept = 0, col = "black")+
  geom_vline(xintercept = 0, col = "black")+
  geom_point(alpha = 0.2)+
  geom_point(data = flysex.stats[select,], aes(x = Ratio.H.L.normalized.0913_QE_IA_NL_FS_125, y = Ratio.H.L.normalized.0913_QE_IA_NL_FS_126), color = "firebrick3")+
  coord_cartesian(xlim = c(-8,8), ylim = c(-8,8))+
  geom_text_repel(
      data = flysex.stats[select == TRUE,],
      aes(x = Ratio.H.L.normalized.0913_QE_IA_NL_FS_125, 
          y = Ratio.H.L.normalized.0913_QE_IA_NL_FS_126, 
          label = Gene.names),
      size = 2,
      box.padding = 0.3,
      point.padding = 0.3) +
   theme_classic()+
  labs(title = "S4A: Rep 1 against rep 2", x = "log2(F/M Rep 1)", y = "log2(F/M Rep 2)")
#dev.off()

# Global look at the data: Gene Set Enrichment Analysis
## Generate the table to upload to WebGestalt
gsea.flysex <- data.frame(flysex.stats$entrez, apply(flysex.stats[,1:2], 1, mean.na))
#write.table(gsea.flysex, "GSEA/Flysex/GSEA.Flysex.deep.rnk", sep = "\t", row.names = FALSE, col.names = FALSE, quote = FALSE)

## GSEA plots: Barchart and volcano plot
gsea.flysex <- read.table("GSEA/Flysex/Flysex_GOBP_deep/enrichment_results_wg_result1526754150.txt", sep = "\t", header = T)
gsea.flysex <- gsea.flysex[c(1:5,11:15),]
gsea.flysex <- gsea.flysex[order(gsea.flysex$NES, decreasing = T),]

### Plot figure S4B. NES is negative as the H/L ratios were reversed to keep the female/male ratio
#cairo_pdf("FigureS3C.pdf")
ggplot(gsea.flysex, aes(x=NES, y=description, label=description)) + 
  geom_point(stat="identity", size=4)  +
  geom_segment(aes(y = description, 
                   x = -NES, 
                   yend = description, 
                   xend = 0))+
  scale_y_discrete(limits=gsea.flysex$description)+
  theme_classic()+
  geom_vline(xintercept = 0, lty = "dotted")+
  labs(title = "S3B: GSEA using Webgestalt on sex specificity", x = "FDR", y = "Categories")
#dev.off()

# Import the Sury 2010 et al. dataset (doi: 10.1074/mcp.M110.000323)
old.sex <- read.table("MaleFemale/Old/IA_FS_our_SILAF_published_SILAF/proteinGroups.txt", header = T, sep = "\t")
old.sex <- old.sex[(old.sex$Reverse != "+"),]
old.sex <- old.sex[(old.sex$Potential.contaminant != "+"),]
old.sex <- old.sex[(old.sex$Only.identified.by.site != "+"),]

## Prepare tables to compare ours and the old dataset
old.sex.red <- data.frame(Protein.ID = old.sex$Protein.IDs, 
                          protein.name = old.sex$Protein.names,
                          gene.name = old.sex$Gene.names, 
                          w1118.F = old.sex$Ratio.H.L.normalized.w1118_female_standard,
                          w1118.M = old.sex$Ratio.H.L.normalized.w1118_male_standard)

old.sex.red$ratio.w118 <- log2(old.sex.red$w1118.M/old.sex.red$w1118.F)

entrez <- select(up, sub(";.*", "", old.sex.red$Protein.ID), "ENTREZ_GENE") #Maps entrez IDs to uniprot IDs
entrez <- entrez[!duplicated(entrez$UNIPROTKB),]
old.sex.red$entrez <- entrez$ENTREZ_GENE[match(entrez$UNIPROTKB, sub(";.*", "", old.sex.red$Protein.ID))]

flysex.comparative <- data.frame(protein.id = old.sex.red$Protein.ID,
                                 protein.name = old.sex$Protein.names,
                                 gene.name = old.sex$Gene.names, 
                                 STHLM.Rep1 = flysex.stats$Ratio.H.L.normalized.0913_QE_IA_NL_FS_125
                                 [match(old.sex.red$entrez, flysex.stats$entrez)],
                                 STHLM.Rep2 = flysex.stats$Ratio.H.L.normalized.0913_QE_IA_NL_FS_126
                                 [match(old.sex.red$entrez, flysex.stats$entrez)],
                                 OLD = old.sex.red$ratio.w118)

flysex.comparative$STHLM.mean <- apply(flysex.comparative[,4:5], 1, mean.na)
flysex.comparative$interest <- abs(flysex.comparative$STHLM.mean) > 1 & abs(flysex.comparative$OLD) > 1

sex.of.interest <- flysex.comparative[(abs(flysex.comparative$STHLM.mean) > 1 & abs(flysex.comparative$OLD) > 1 ),]
sex.of.interest <- sex.of.interest[!is.na(sex.of.interest$protein.id),]

#Calculate R:
round(cor(flysex.comparative$STHLM.mean, flysex.comparative$OLD, use = "complete.obs"), digits=4)

## Figure S3C
#cairo_pdf("FigureS3D.pdf")
ggplot(data = flysex.comparative, aes (x = STHLM.mean, y = OLD))+
  geom_point(alpha = 0.3)+
  geom_point(data = sex.of.interest, color = "red")+
  geom_abline(intercept = 0, slope = 1)+
  geom_text_repel(data = sex.of.interest,
    aes(x = STHLM.mean, y = OLD, label = gene.name),
    size = 3,
    box.padding = 0.5,
    point.padding = 0.5)+
  theme_classic()+
  geom_smooth(method='lm',formula=y~x)+
  coord_cartesian(xlim = c(-10,10), ylim = c(-10,10))
#dev.off()
```

# Holidic to yeast food proteome comparison: Figure 2 and S3
To what extent is the fly grown on holidic food differing from yeast grown flies?

## Differential proteome

```{r Holidic_Proteome, tidy = TRUE, warning = FALSE}
library("tidyverse")
holidic.proteome <- read.table("Holidic/proteinGroups.txt", header = T, sep = "\t")

## Filtering
## Filtering
holidic.proteome <- holidic.proteome[!holidic.proteome$Potential.contaminant == "+",]
holidic.proteome <- holidic.proteome[!holidic.proteome$Reverse == "+",]
holidic.proteome <- holidic.proteome[!holidic.proteome$Only.identified.by.site == "+",]

rownames(holidic.proteome) <- holidic.proteome$Protein.IDs

holidic.proteome.red <- data.frame(ID = holidic.proteome$Protein.IDs,
                                   protein = holidic.proteome$Protein.names,
                                   gene = holidic.proteome$Gene.names,
                                   Rep1 = log2(holidic.proteome$Ratio.H.L.normalized.1),
                                   Rep2 = log2(holidic.proteome$Ratio.H.L.normalized.2))


holidic.proteome.complete <- holidic.proteome.red[complete.cases(holidic.proteome.red[,4:5]),]
rownames(holidic.proteome.complete) <- holidic.proteome.complete$ID

holidic.proteome.complete$Ratio <- apply(holidic.proteome.complete[,4:5], 1, mean.na)

## Calculate padj with limma moderated t-test
fit <- lmFit(holidic.proteome.complete[,4:5])
fit <- eBayes(fit)
limma.result <- topTable(fit, number = Inf, confint = TRUE, adjust.method = "fdr")

holidic.proteome.complete$limma.padj <- limma.result[match(holidic.proteome.complete$ID, rownames(limma.result)),]$adj.P.Val

## Prepare a table for GSEA
holidic.proteome.red$Ratio <- apply(holidic.proteome.red[,4:5],1,mean.na)
holidic.proteome.red <- holidic.proteome.red[!is.na(holidic.proteome.red$Ratio),] 

rankedSet <- select(up, sub(";.*","", holidic.proteome.red$ID), "ENTREZ_GENE")

monster.gsea <- data.frame(entrez = rankedSet$ENTREZ_GENE, 
                           logfc = holidic.proteome.red[match(rankedSet$UNIPROTKB, sub(";.*","", holidic.proteome.red$ID)),]$Ratio)
monster.gsea <- subset(monster.gsea, !duplicated(monster.gsea$entrez))
#write.table(monster.gsea, "GSEA/Monster/Monster.deep.rnk", quote = F, col.names = F, row.names = F, sep = "\t")

##Prepare a table for ORA
holidic.proteome.complete$ENTREZ <- mapIds(org.Dm.eg.db, 
                               keys=as.character(holidic.proteome.complete$ID), 
                               column="ENTREZID", 
                               keytype="UNIPROT",
                               multiVals="first")

monster.ora <- holidic.proteome.complete[holidic.proteome.complete$limma.padj < 0.001,]$ENTREZ
#write.table(monster.ora, file = "GSEA/Monster/Monster_ORA.deep.txt", col.names = F, quote = F,
            row.names = F)


## Import the WebGestalt output and plot
monster.gsea.webgestalt <- read.table("GSEA/Monster/Proteome_deep_Results/Monster.proteome.GSEA.results.txt",
                           header = T, sep = "\t")

## A color palette with grey:
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#0072B2", "#D55E00", "#CC79A7")

## Figure S6A
#cairo_pdf("Figure2C.pdf")
ggplot(data = monster.gsea.webgestalt, aes(x = NES, y = -log10(padj), color = Reference, size = coverage))+
  geom_point()+
  theme_classic()+
  geom_text_repel(aes(label = Category),
                  box.padding = 0.5,
                  point.padding = 0.5)+
  scale_colour_manual(values=cbPalette)
#dev.off()

## How many proteins were calculated in either replicate?
table(!is.na(holidic.proteome.red$Ratio))

# Figure 2A
## GSEA identifies "KEGG Fatty acid metabolism" as a major downregulated category. The names of the candidates come with the Webgestalt output and will be highlighted in a volcano plot

FAM <- c("40692","38457","37446","44117","43591","35213","35761","33524","34614","39860","31703","117361","43592","117369","3355111")
Glyco <- c("35886","43582","45880","43183","42620","33351","43191","33461","44008","34021","42185","39988","45875","31406","36060","43437","43447","36020", "37131", "36020")

#holidic.proteome.complete$entrez <- rankedSet[match(sub(";.*","", holidic.proteome.complete$ID), rankedSet$UNIPROTKB),]$ENTREZ_GENE

holidic.proteome.complete$entrez <- mapIds(org.Dm.eg.db, 
                                     keys=as.character(sub(";.*","", holidic.proteome.complete$ID)), 
                                     column="ENTREZID", 
                                     keytype="UNIPROT",
                                     multiVals="first")


holidic.proteome.complete$FAM <- holidic.proteome.complete$entrez %in% FAM
holidic.proteome.complete$Glyco <- holidic.proteome.complete$entrez %in% Glyco

#cairo_pdf("Figure2A.pdf")
ggplot(data = holidic.proteome.complete, aes(x = Ratio, y = -log10(limma.padj))) +
    geom_point(data = holidic.proteome.complete[holidic.proteome.complete$FAM == FALSE&holidic.proteome.complete$Glyco == FALSE,],
               color = "grey", alpha = 0.6)+
    geom_point(data = holidic.proteome.complete[holidic.proteome.complete$FAM == TRUE,], 
               color = "brown", alpha = 1)+
  geom_point(data = holidic.proteome.complete[holidic.proteome.complete$Glyco == TRUE,], 
               color = "cornflowerblue", alpha = 1)+
    theme_classic()+
    labs(x = "log2(fold change)", y = "-log10(adjusted p-value)")+
    geom_hline(yintercept = -log10(0.001), color = "black", lty = "dotted")+
    geom_text_repel(data = holidic.proteome.complete[holidic.proteome.complete$Glyco == TRUE,], aes(label = gene))+
  geom_text_repel(data = holidic.proteome.complete[holidic.proteome.complete$FAM == TRUE,], aes(label = gene))+
    coord_cartesian(xlim = c(-4,4))
#dev.off()

#write.table(holidic.proteome.complete, ".../SILAF/Submission/Holidic_Proteome.txt",  sep = "\t")

```

## Differential phosphorylome: Figure 2C and figure S5B-E

```{r Phosphorylome, tidy = TRUE, warning = FALSE}
# Start by looking at the phospho sites
phospho <- read.csv(".../SILAF/MQ results/SILAF/Markdown/Phospho/Phospho(STY)Sites.csv", 
                    header = T, sep =";")


# Exclusion criterion
phospho <- phospho[phospho$Localization.prob>=0.75,]

# Prepare the table for Supplementary table S2
publication <- data.frame(protein.id = phospho$Protein,
                          protein = phospho$Protein.names,
                          gene.name = phospho$Gene.names,
                          position = phospho$Positions.within.proteins,
                          seq = phospho$Sequence.window,
                          Ratio.1 = phospho$Ratio.H.L.normalized.1,
                          Ratio.1 = phospho$Ratio.H.L.normalized.2,
                          occupancy.yeast = phospho$Occupancy.L,
                          occupancy.holidic = phospho$Occupancy.H)
publication$occupancy.diff <- log2(publication$occupancy.holidic/publication$occupancy.yeast)

## How is the H/L ratio per site of the two replicates towards each other? Figure S6C

#Calculate R:
round(cor(log2(phospho$Ratio.H.L.normalized.1), log2(phospho$Ratio.H.L.normalized.2), use = "complete.obs"), digits=4)

#cairo_pdf("FigureS3B.pdf")
ggplot(data = phospho, aes(x = log2(Ratio.H.L.normalized.1), y = log2(Ratio.H.L.normalized.2)))+
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  geom_abline(slope = 1, intercept = 0, lty = "dotted")+
  theme_classic()+
  geom_point(data = subset(phospho, abs(log2(Ratio.H.L.normalized.1)) < 2 &
                             abs(log2(Ratio.H.L.normalized.2)) < 2),
             color = "black", alpha = 0.3)+
  geom_point(data = subset(phospho, abs(log2(Ratio.H.L.normalized.1)) >= 2 &
                             abs(log2(Ratio.H.L.normalized.2)) >= 2),
             color = "firebrick3", alpha = 1)+
  geom_text_repel(
    data = subset(phospho, abs(log2(Ratio.H.L.normalized.1)) >= 2 &
                             abs(log2(Ratio.H.L.normalized.2)) >= 2),
    aes(label = Gene.names),
    size = 5,
    box.padding = 0.5,
    point.padding = 0.5)+
  coord_fixed(ratio = 1/1)
#dev.off()

## Looking at occupancies: individual replicates
phospho.occupancy <- data.frame(gene = phospho$Gene.names,
                                protein = phospho$Proteins,
                                name = phospho$Protein.names,
                                Rep1 = log2(phospho$Occupancy.H.1/phospho$Occupancy.L.1), 
                                Rep2 = log2(phospho$Occupancy.H.2/phospho$Occupancy.L.2),
                                Rep.comb = log2(phospho$Occupancy.H/phospho$Occupancy.L),
                                locProb = phospho$Localization.prob,
                                mean.MQ = log2(phospho$Occupancy.H/phospho$Occupancy.L))

phospho.occupancy$mean <- apply(phospho.occupancy[,4:5],1,mean.na)
phospho.occupancy <- phospho.occupancy[complete.cases(phospho.occupancy$mean),]

### Figure S3C:
#cairo_pdf("FigureS3C.pdf")
ggplot(data = phospho.occupancy, aes(x = Rep1, y = Rep2))+
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  geom_abline(slope = 1, intercept = 0, lty = "dotted")+
  theme_classic()+
  geom_point(data = phospho.occupancy[abs(phospho.occupancy$Rep1) < 1 & abs(phospho.occupancy$Rep2) < 1,],
             color = "black", alpha = 0.3)+
  geom_point(data = phospho.occupancy[abs(phospho.occupancy$Rep1) >= 1 & 
                                        abs(phospho.occupancy$Rep2) >= 1,],
             color = "firebrick3", alpha = 1)+
  geom_text_repel(
    data = phospho.occupancy[abs(phospho.occupancy$Rep1) >= 1 & 
                                        abs(phospho.occupancy$Rep2) >= 1,],
      aes(x = Rep1, y = Rep2, label = gene),
      size = 3,
      box.padding = 0.3,
      point.padding = 0.3)+
  coord_fixed(ratio = 1/1)
#dev.off()

# Calculate R:
round(cor(phospho.occupancy$Rep1, phospho.occupancy$Rep2, use = "complete.obs"), digits=4)

### Further analysed in the online module of STRING

## Global approach to the data with a gene set enrichment analysis. As a GSEA is not robust against multiple entries per protein/gene, this analysis takes the max/min value per entrez ID into account

### Prepare a ranked list
genes.tmp <- unlist(strsplit(as.character(phospho.occupancy$protein), "[;]"))
counter.tmp <- str_count(phospho.occupancy$protein, ";")+1 # Split the protein IDs

phospho.gsea <- data.frame(uniprot = genes.tmp, logfc = rep(phospho.occupancy$mean, counter.tmp))
rankedSet <- select(up, phospho.gsea$uniprot, "ENTREZ_GENE")
phospho.gsea <- data.frame(entrez = rankedSet$ENTREZ_GENE, 
                           logfc = phospho.gsea[match(rankedSet$UNIPROTKB, phospho.gsea$uniprot),]$logfc)
phospho.gsea <- phospho.gsea[order(abs(phospho.gsea$logfc)),] # As to remove duplicates in the next step, keep the maximum phospho occupancy change

phospho.gsea <- subset(phospho.gsea, !duplicated(phospho.gsea$entrez))
#write.table(file = "GSEA/Monster/Monster.phospho.rnk", phospho.gsea, sep = "\t", col.names = F, 
#            row.names = F, quote = F)

### Import the WebGestalt output and plot
phospho.gsea.results <- read.table("GSEA/Monster/Phospho_Results/Monster.phospho.GSEA.results.txt",
                           header = T, sep = "\t")

### A color palette with grey:
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#0072B2", "#D55E00", "#CC79A7")

### Figure S6B
#cairo_pdf("FigureS3D.pdf")
ggplot(data = phospho.gsea.results, aes(x = NES, y = -log10(padj), color = Reference, size = coverage))+
  geom_point()+
  theme_classic()+
  geom_text_repel(aes(label = Category),
                  box.padding = 0.5,
                  point.padding = 0.5)+
  scale_colour_manual(values=cbPalette)
#dev.off()

## Looking at occupancies: cummulative values
phospho.occupancy.cuml <- data.frame(protein = phospho$Proteins,
                                      name = phospho$Protein.names,
                                      gene = phospho$Gene.names,
                                      occupancy = log2(phospho$Occupancy.H/phospho$Occupancy.L),
                                      Occupancy.H = phospho$Occupancy.H,
                                      Occupancy.L = phospho$Occupancy.L, 
                                      PEP = phospho$PEP,
                                      seq = phospho$Sequence.window)

### Exclusion criteria
phospho.occupancy.cuml <- phospho.occupancy.cuml[phospho.occupancy.cuml$occupancy!= "NaN",]
phospho.occupancy.cuml <- phospho.occupancy.cuml[!is.na(phospho.occupancy.cuml$seq),]

### STRING of highly changed targets
string.changed <- as.character(phospho.occupancy.cuml[abs(phospho.occupancy.cuml$occupancy) >= 1,]$protein)
string.up <- as.character(phospho.occupancy.cuml[phospho.occupancy.cuml$occupancy >= 1,]$protein)
string.down <- as.character(phospho.occupancy.cuml[phospho.occupancy.cuml$occupancy <= -1,]$protein)
string.background <- as.character(phospho.occupancy.cuml$protein)
#write.table(file = "GSEA/Monster/Phosphostring_changed.txt", string.changed, quote = F, col.names = F, row.names = F)
#write.table(file = "GSEA/Monster/Phosphostring_up.txt", string.up, quote = F, col.names = F, row.names = F)
#write.table(file = "GSEA/Monster/Phosphostring_down.txt", string.down, quote = F, col.names = F, row.names = F)
#write.table(file = "GSEA/Monster/Phosphostring_bg.txt", string.background, quote = F, col.names = F, row.names = F)

### Plot occupancies in a histogram
#pdf("Figure2D.pdf")
ggplot(data = phospho.occupancy.cuml, aes(occupancy))+
  geom_histogram()+
  theme_classic()
#dev.off()

dim(phospho[phospho$Localization.prob>=0.75,])
  
# How many proteins are there in total? Basis for Figure 2B
df <- data.frame(x = c(3,6,9,12), y = c(3,6,9,12))
size <- c(12221, 7496, 1356, 629)
size <- sqrt(size/pi)*2

#cairo_pdf("Figure2B1.pdf")
ggplot(data = df, aes (x = x, y = y))+
  geom_point(size = size)+
  coord_cartesian(xlim=c(-2,15), ylim = c(-2,15))
#dev.off()

df <- data.frame(x = c(3,6,9,12), y = c(3,6,9,12))
size <- c(5966, 4749, 3880, 141)
size <- sqrt(size/pi)*2

#cairo_pdf("Figure2B2.pdf")
ggplot(data = df, aes (x = x, y = y))+
  geom_point(size = size)+
  coord_cartesian(xlim=c(-2,15), ylim = c(-2,15))
#dev.off()

# Sequence motives: FigureS3A
## Global

t <- phospho.occupancy.cuml[!is.na(phospho.occupancy.cuml$seq),]

#pdf("FigureS3A.pdf")
ggseqlogo(sub(";.*","",as.character(t$seq)), method = 'bits', col_scheme='chemistry')
#dev.off()

## Testing the environment of specific phospho AA
ggseqlogo(sub(";.*","",as.character(t[substr(as.character(t$seq), 16, 16)=="Y",]$seq)), method = 'bits', col_scheme='chemistry')

# Depicting selected targets
## Groups taken from the earlier GSEA
gluco.phospho <- c("Q9XTL9", "P52034", "Q9VM14")
proteasome.phospho <- rankedSet[match(c("35701","37029","39628"), rankedSet$ENTREZ_GENE),]$UNIPROTKB
FGF.phospho <- rankedSet[match(c("42186","42366","41020","48311","44801"), rankedSet$ENTREZ_GENE),]$UNIPROTKB
translation.phospho <- rankedSet[match(c("41347","42761","36985","39877","39386","43594","40451","34329","40563","37430","36271","31897","38983","34352","32635","37292","39028","34363","34309","43768","33487","43349","35422"),rankedSet$ENTREZ_GENE),]$UNIPROTKB
manual.phospho  <- c("P11995", "P45437")

## How often is one protein phosphorylated?
length.gluco <- length(phospho[which(sub(";.*","",phospho$Proteins) %in% c(gluco.phospho)),]$Proteins)
length.FGF <- length(phospho[which(sub(";.*","",phospho$Proteins) %in% c(FGF.phospho)),]$Proteins)
length.translation <- length(phospho[which(sub(";.*","",phospho$Proteins) %in% c(translation.phospho)),]$Proteins)
length.proteasome <- length(phospho[which(sub(";.*","",phospho$Proteins) %in% c(proteasome.phospho)),]$Proteins)
length.manual <- length(phospho[which(sub(";.*","",phospho$Proteins) %in% c(manual.phospho)),]$Proteins)

## Create on data.frame
phospho.select.gluco <- phospho[which(sub(";.*","",phospho$Proteins) %in% c(gluco.phospho)),]
phospho.select.FGF <- phospho[which(sub(";.*","",phospho$Proteins) %in% c(FGF.phospho)),]
phospho.select.translation <- phospho[which(sub(";.*","",phospho$Proteins) %in% c(translation.phospho)),]
phospho.select.proteasome <- phospho[which(sub(";.*","",phospho$Proteins) %in% c(proteasome.phospho)),]
phospho.select.manual <- phospho[which(sub(";.*","",phospho$Proteins) %in% c(manual.phospho)),]

phospho.select <- rbind(phospho.select.gluco, phospho.select.FGF, phospho.select.translation, phospho.select.proteasome, phospho.select.manual)

phospho.select <- data.frame(name = rep(phospho.select$Gene.names,2), uniprot = rep(phospho.select$Proteins,2),
                             occupancy = c(phospho.select$Occupancy.L, phospho.select$Occupancy.H),
                             type = c(rep("L", length(phospho.select$Gene.names)), 
                                      rep("H", length(phospho.select$Gene.names))),
                             seq = c(1:(length(phospho.select$Gene.names))),
                             site = paste(phospho.select$Amino.acid, phospho.select$Positions.within.proteins, sep = ""),
                             category = c(rep("3gluco", length.gluco), rep("4FGF", length.FGF), rep("2translation", length.translation), rep("1proteasome", length.proteasome), 
                                          rep("0manual", length.manual)))

pdh <- data.frame(name = c("Pdh-E2","Pdh-E2"),
                  uniprot = c("Q9VM14", "Q9VM14"),
                  occupancy = c(0.1421, 0.0092753),
                  type = c("H", "L"),
                  seq = c(length(phospho.select$seq)+1, length(phospho.select$seq)+2),
                  site = c("S252", "S252"),
                  category = c("0manual", "0manual"))
                  
phospho.select <- rbind(phospho.select, pdh)

phospho.select <- phospho.select[with(phospho.select, order(category, type, uniprot, decreasing = T)),]

phospho.select$name.unique <- paste(phospho.select$name, " ", phospho.select$site, sep = "") # Give each phospho site a unique ID
phospho.select$name.unique <- factor(phospho.select$name.unique, levels = unique(phospho.select$name.unique))
phospho.select$type <- factor(phospho.select$type, levels = unique(phospho.select$type))

phospho.select <- phospho.select[phospho.select$occupancy != 0,]
phospho.select <- phospho.select[!is.na(phospho.select$occupancy),]

## Figure S3E
#cairo_pdf("FigureS3E.pdf", 10, 4)
ggplot(data = phospho.select, aes(x=name.unique, y = occupancy, fill = type))+
  geom_bar(stat="identity", position=position_dodge())+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_fill_manual(values=c("grey", "firebrick3"))+
  coord_cartesian(ylim= c(0,1))
#dev.off()
```

# Mitochondrial proteome and phosphorylome upon loss of LRPPRC

## LRPPRC mitoproteome

```{r LRPPRC_phospho total proteome}
###############################
### Total proteome overview ###
###############################

dm.mitocarta <- read.csv("deepLRPPRC_Fly/Fly_Interactome_curated.csv")
oxphos.library <- read.table("deepLRPPRC_Fly/OXPHOS_components_Mm.Dm.txt", header = T, sep = "\t") # Manually curated last: 190702

## Prepare the data frame
lrpprc.deep <- read_tsv('deepLRPPRC_Fly/IA_FS_LRPPRC_SILAF_Phospho_with_match_06_2019/data/proteinGroups.txt')
lrpprc.deep.HL <- log2(lrpprc.deep[,grepl("Ratio.H.L.normalized.Exp_", colnames(lrpprc.deep))]) #Focus on normalized H/L ratios and log2 convert
lrpprc.deep.HL[,c(2,4)] <- -lrpprc.deep.HL[,c(2,4)] #Turn around the label swaps
rownames(lrpprc.deep.HL) <- lrpprc.deep$`Majority protein IDs`

## Limma stats
fit <- lmFit(lrpprc.deep.HL)
fit <- eBayes(fit)
limma.result <- topTable(fit,number = Inf,confint = TRUE, adjust.method = "fdr")

## Annotation
limma.result$protein <- sub(";.*","", rownames(limma.result))

limma.result$ENSEMBL <- mapIds(org.Dm.eg.db, 
                               keys=as.character(limma.result$protein), 
                               column="ENSEMBL", 
                               keytype="UNIPROT",
                               multiVals="first")

limma.result$SYMBOL <- as.character(mapIds(org.Dm.eg.db, 
                                           keys=as.character(limma.result$protein), 
                                           column="SYMBOL", 
                                           keytype="UNIPROT",
                                           multiVals="first"))

limma.result$dm.mitocarta <- dm.mitocarta[match(limma.result$ENSEMBL, dm.mitocarta$Gene.ID),]
limma.result$mitochondrial <- limma.result$ENSEMBL %in% dm.mitocarta$Gene.ID

## Global volcano plot
ggplot(data = limma.result, aes(x = logFC, y = -log10(adj.P.Val)))+
  geom_point()+
  geom_point(data = limma.result[limma.result$mitochondrial == TRUE,], aes(x = logFC, y = -log10(adj.P.Val)), 
             col = "red")+
  geom_label_repel(data = limma.result[limma.result$adj.P.Val < 0.001,], aes(x = logFC, y = -log10(adj.P.Val), 
                                                                             label = SYMBOL, col = mitochondrial))+
  scale_color_manual(values=c("black", "red"))+
  theme_bw()+
  labs(title = "LRPPRC knockdown: limma on normalized H/L ratios")

total.proteome.after.enrich <- limma.result # Assign unambigous variable for later

#write.table(total.proteome.after.enrich, "deepLRPPRC_Fly/LRPPRC_proteome.for.STable.txt", sep = "\t")


## Enrichment
lrpprc.deep$ENSEMBL <-  mapIds(org.Dm.eg.db, 
                               keys=as.character(sub(";.*","", lrpprc.deep$`Majority protein IDs`)), 
                               column="ENSEMBL", 
                               keytype="UNIPROT",
                               multiVals="first")

lrpprc.deep$mito <- lrpprc.deep$ENSEMBL %in% dm.mitocarta$Gene.ID

#cairo_pdf("deepLRPPRC_Fly/FigureS4A_Mito.enrichment_LRPPRC.pdf", 3, 3)
ggplot(data = lrpprc.deep, aes (x = mito, y = log2(Intensity)))+
  geom_boxplot()+
  coord_cartesian(ylim=c(15,45))+
  theme_classic()
#dev.off()

holidic.proteome$ENSEMBL <- mapIds(org.Dm.eg.db, 
                               keys=as.character(sub(";.*","", holidic.proteome$Majority.protein.IDs)), 
                               column="ENSEMBL", 
                               keytype="UNIPROT",
                               multiVals="first")

holidic.proteome$mito <- holidic.proteome$ENSEMBL %in% dm.mitocarta$Gene.ID

#cairo_pdf("deepLRPPRC_Fly/FigureS4A_Mito.enrichment_Holidic.pdf", 3, 3)
ggplot(data = holidic.proteome, aes (x = mito, y = log2(Intensity)))+
  geom_boxplot()+
  coord_cartesian(ylim=c(15,45))+
  theme_classic()
#dev.off()

median(holidic.proteome[holidic.proteome$mito %in% TRUE,]$Intensity)/
  median(holidic.proteome[holidic.proteome$mito %in% FALSE,]$Intensity)

## Mitosubsetting
total <- total.proteome.after.enrich
total <- total[!is.na(total$logFC),]
total <- total[!is.na(total$ENSEMBL),] # Eliminates reverse and contaminants

### Categorical t-test with fdr correction and boxplotting
cat.t.test <- function(data, abs = "FALSE"){
  data <- total
  cats <- unique(data$dm.mitocarta$Mitochondrial.Process)
  cats = cats[!is.na(cats)]
  p <- c()
  for(i in c(1:length(cats))){
    data.sub <- data[which(data$dm.mitocarta$Mitochondrial.Process == cats[i]),]
    if(dim(data.sub)[1] <= 1){
      p.add <- NA
    } else {
      p.add <- p.adjust(unlist(t.test(data.sub$logFC)[3]), method = "fdr", n = length(data.sub$logFC))
      if(abs == "TRUE"){
        p.add <- p.adjust(unlist(t.test(abs(data.sub$logFC))[3]), method = "fdr", n = length(data.sub$logFC))
      }
    }
    p <- c(p,p.add)
  }
  stats <- data.frame(Category = cats, p.value = round(p,20))
  return(stats)
}

cat.t.test(total, abs = "FALSE")

#cairo_pdf("deepLRPPRC_Fly/Figure3A_Boxplot.categories.pdf", 10, 4)
ggplot(total[!is.na(total$dm.mitocarta$Mitochondrial.Process),], 
       aes(x = fct_reorder(dm.mitocarta$Mitochondrial.Process, logFC, .desc = TRUE), y = logFC))+
  geom_hline(yintercept = 0)+
  theme_bw()+
  geom_boxplot(outlier.size = 0.2)+
  #coord_cartesian(ylim = c(-3,3))+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  geom_jitter(position=position_jitter(0.2), size = 0.2)+
  labs(title = "Mitochondrial functional categories in LRPPRC proteomics")
#dev.off()

### Heatmapping by complex
proc <- lrpprc.deep.HL
proc$ENSEMBL <- mapIds(org.Dm.eg.db, 
                       keys=sub(";.*","", rownames(proc)), 
                       column="ENSEMBL", 
                       keytype="UNIPROT",
                       multiVals="first")

proc <- proc[!is.na(proc$ENSEMBL),]
proc$complex <- oxphos.library[match(proc$ENSEMBL, oxphos.library$Dm),]$Complex
proc <- proc[order(match(proc$ENSEMBL,oxphos.library$Dm)),]
proc$category <- dm.mitocarta[match(proc$ENSEMBL, dm.mitocarta$Gene.ID),]

#etc.members <- proc[proc$category$Mitochondrial.Process %in% "Electron Transport Chain",]
#write.table(etc.members, file = "deepLRPPRC_Fly/Missing.ETC.components.txt", sep = "\t")

proc <- proc[!is.na(proc$complex),]
proc <- proc[!rowSums(is.na(proc[,1:4])) >= 4,] # Exclude all four values NA
proc$homolog <- oxphos.library[match(proc$ENSEMBL, oxphos.library$Dm),]$Mm

proc <- proc[with(proc, order(complex, homolog)),]

total.oxphos.heatmap <- proc[,1:4]
total.oxphos.heatmap <- total.oxphos.heatmap[,order(c(1,3,2,4))] # Aesthetically, turn around the order of samples
rownames(total.oxphos.heatmap) <- paste(proc$complex, proc$ENSEMBL, proc$homolog, sep = ";")
total.oxphos.heatmap[total.oxphos.heatmap == "NaN"] <- NA

breakList <- seq(-2.1,2.1, by = 0.25)
colors <- colorpanel(length(breakList)-1, "deepskyblue", "black", "yellow")

#cairo_pdf("deepLRPPRC_Fly/Figure3B.FigureS4D_Total.prot.pheatmap.pdf", 5, 65)
pheatmap(total.oxphos.heatmap, color = colors, breaks = breakList, cluster_cols = F, cluster_rows = F,
         main = "LRPPRC fold changes per replicate", gaps_row = c(37,40,50,62,80,81,95,109,113,117,133))
#dev.off()

### Supplementary bsf levels
bsf.levels <- lrpprc.deep[lrpprc.deep$`Gene names` %in% "bsf",]
bsf.levels <- bsf.levels[,grepl("Intensity.[LH].", colnames(bsf.levels))]
bsf.levels <- data.frame(t(bsf.levels))
bsf.levels$condition <- rep(c("C", "KD", "KD", "C"),2)

#cairo_pdf("deepLRPPRC_Fly/FigureS4A_LRPPRC_changes.pdf")
ggplot(data = bsf.levels, aes(x = condition, y = log2(t.bsf.levels.)))+
  geom_point()+
  theme_classic()+
  coord_cartesian(ylim=c(15,45))+
  theme_classic()
#dev.off()

### Supplementary GSEA O-term enrichment
gsea.bsf <- data.frame(name = total.proteome.after.enrich$ENSEMBL, FC = total.proteome.after.enrich$logFC)
#write.table(gsea.bsf, file = "deepLRPPRC_Fly/GSEA.bsf.in.rnk", row.names = F, col.names = F, quote = F, sep = "\t")

webgestalt.bsf <- read.table("deepLRPPRC_Fly/Webgestalt_bsf.total_affinity.prop_KEGG.Reactome.txt", sep = "\t")
colnames(webgestalt.bsf) <- c("Name", "NES")

webgestalt.bsf$Name <- factor(webgestalt.bsf$Name, levels = webgestalt.bsf$Name)

#pdf("deepLRPPRC_Fly/FigureS4B_Webgestalt_bsf.total_affinity.prop_KEGG.Reactome.pdf")
ggplot(data = webgestalt.bsf, aes (y = NES, x = Name))+
  geom_bar(stat = "identity", col = "black")+
  theme_classic()
#dev.off()

## Overlap fly and mouse

### Import the data from Kühl et al. (2017)
kuehl.lrpprc <- read.table("LRPPRC_Mouse/LRPPRC_Mouse.txt", sep = "\t", header = T)

fly <- total.proteome.after.enrich

## Make a new data frame overlapping fly and mouse data
sthlm.kuehl.lrpprc <- data.frame(name = oxphos.library$SYMBOL,
                                 Dm = fly[match( oxphos.library$Dm, fly$ENSEMBL),]$logFC,
                                 Mm.20 = kuehl.lrpprc[match(oxphos.library$Mm, kuehl.lrpprc$Gene.names),]$logFC.2weeks,
                                 Mm.34 = kuehl.lrpprc[match(oxphos.library$Mm, kuehl.lrpprc$Gene.names),]$logFC.3.4weeks,
                                 Mm.50 = kuehl.lrpprc[match(oxphos.library$Mm, kuehl.lrpprc$Gene.names),]$logFC.5weeks,
                                 Mm.70 = kuehl.lrpprc[match(oxphos.library$Mm, kuehl.lrpprc$Gene.names),]$logFC.7weeks,
                                 Mm.99 = kuehl.lrpprc[match(oxphos.library$Mm, kuehl.lrpprc$Gene.names),]$logFC.10weeks,
                                 complex = oxphos.library$Complex)

sthlm.kuehl.lrpprc <- gather(sthlm.kuehl.lrpprc, "Species", "Intensity", 2:7)

sthlm.kuehl.lrpprc <- sthlm.kuehl.lrpprc[sthlm.kuehl.lrpprc$complex == "1" |
                                         sthlm.kuehl.lrpprc$complex == "2" |
                                         sthlm.kuehl.lrpprc$complex == "3" |
                                         sthlm.kuehl.lrpprc$complex == "4" |
                                         sthlm.kuehl.lrpprc$complex == "5",]

#cairo_pdf("FigureS4E_LRPPRC_Kuehl_Sthlm.pdf", 7, 4)
ggplot(data = sthlm.kuehl.lrpprc, aes(x = complex, y = Intensity, fill = Species))+
  geom_boxplot()+
  theme_classic()+
  scale_fill_brewer(palette = "Purples")
#dev.off()

```

## LRPPRC mitophosphorylome

```{r LRPPRC_phospho phosphorylome}
#####################
### Phosphorylome ###
#####################

# Prepare data and filter
bsf.phospho.match <- read_tsv("deepLRPPRC_Fly/IA_FS_LRPPRC_SILAF_Phospho_with_match_06_2019/data/Phospho (STY)Sites.txt")

bsf.phospho.match <- bsf.phospho.match[!grepl("REV__",bsf.phospho.match$Protein),]
bsf.phospho.match <- bsf.phospho.match[bsf.phospho.match$`Localization prob` >= 0.75,]

rownames(bsf.phospho.match) <- paste(bsf.phospho.match$`Leading proteins`, ";", bsf.phospho.match$`Positions within proteins`, sep = "")

Occupancy.L <- bsf.phospho.match[,grepl("Occupancy.L", colnames(bsf.phospho.match))]
Occupancy.L[Occupancy.L == 0] <- NA # Assuming that an occupancy of 0 doesn't exist if the site is picked up
Occupancy.H <- bsf.phospho.match[,grepl("Occupancy.H", colnames(bsf.phospho.match))]
Occupancy.H[Occupancy.L == 0] <- NA # again

Occupancy <- log2(Occupancy.H/Occupancy.L)
Occupancy[,c(3,5)] <- -Occupancy[,c(3,5)]
Occupancy <- Occupancy[,-1]

rownames(Occupancy) <- rownames(bsf.phospho.match)
rownames(Occupancy.L)<- rownames(bsf.phospho.match)
rownames(Occupancy.H)<- rownames(bsf.phospho.match)

# Stats with limma
fit <- lmFit(Occupancy)
fit <- eBayes(fit)
limma.result <- topTable(fit,number = Inf,confint = TRUE, adjust.method = "fdr")

limma.result$protein <- sub(";.*","", rownames(limma.result))

limma.result$ENSEMBL <- mapIds(org.Dm.eg.db, 
                               keys=as.character(limma.result$protein), 
                               column="ENSEMBL", 
                               keytype="UNIPROT",
                               multiVals="first")

limma.result$SYMBOL <- as.character(mapIds(org.Dm.eg.db, 
                                           keys=as.character(limma.result$protein), 
                                           column="SYMBOL", 
                                           keytype="UNIPROT",
                                           multiVals="first"))

limma.result$dm.mitocarta <- dm.mitocarta[match(limma.result$ENSEMBL, dm.mitocarta$Gene.ID),]
limma.result$mitochondrial <- limma.result$ENSEMBL %in% dm.mitocarta$Gene.ID
limma.result$OXPHOS <- limma.result$ENSEMBL %in% oxphos.library$Dm

limma.result$occ.L <- Occupancy.L[match(rownames(limma.result), rownames(Occupancy.L)),2:5]
limma.result$occ.H <- Occupancy.H[match(rownames(limma.result), rownames(Occupancy.H)),2:5]

limma.result$mean.occ.ctrl <- apply(data.frame(limma.result$occ.L$`Occupancy L Exp_01`,
                                     limma.result$occ.L$`Occupancy L Exp_03`,
                                     limma.result$occ.H$`Occupancy H Exp_02`,
                                     limma.result$occ.H$`Occupancy H Exp_04`), 1, mean.na)
limma.result$mean.occ.KD <- apply(data.frame(limma.result$occ.L$`Occupancy L Exp_02`,
                                     limma.result$occ.L$`Occupancy L Exp_04`,
                                     limma.result$occ.H$`Occupancy H Exp_01`,
                                     limma.result$occ.H$`Occupancy H Exp_03`), 1, mean.na)

limma.result$peptide <- bsf.phospho.match[match(rownames(limma.result), rownames(bsf.phospho.match)),]$`Phospho (STY) Probabilities`
limma.result$PEP <- bsf.phospho.match[match(rownames(limma.result), rownames(bsf.phospho.match)),]$PEP
limma.result$ratio.occ <- Occupancy[match(rownames(limma.result), rownames(Occupancy)),]
limma.result$position <- bsf.phospho.match[match(rownames(limma.result), rownames(bsf.phospho.match)),]$`Positions within proteins`

#write.table(limma.result, "deepLRPPRC_Fly/LRPPRC_phosphorylome.for.STable.txt", sep = "\t")

# Volcano plotting mitochondrial for supplementary
#cairo_pdf("deepLRPPRC_Fly/Figure4C_Phospho_Occupancies_mito.pdf", 6, 6)
ggplot(data = limma.result, aes(x = logFC, y = -log10(adj.P.Val)))+
  geom_point(data = limma.result[limma.result$mitochondrial != TRUE,], aes(x = logFC, y = -log10(adj.P.Val)), 
             col = "grey")+
  geom_point(data = limma.result[limma.result$mitochondrial %in% TRUE &
                                   limma.result$dm.mitocarta$Mitochondrial.Process != 
                                   "Electron Transport Chain",], aes(x = logFC, y = -log10(adj.P.Val)), 
             col = "black")+
  geom_text_repel(data = limma.result[limma.result$adj.P.Val < 0.05 & limma.result$mitochondrial == TRUE &
                                        limma.result$dm.mitocarta$Mitochondrial.Process != 
                                        "Electron Transport Chain",], 
                  aes(x = logFC, y = -log10(adj.P.Val), 
                      label = SYMBOL), color = "black", force = 10, max.iter = 5000)+
  geom_text_repel(data = limma.result[limma.result$adj.P.Val < 0.05 & limma.result$mitochondrial != TRUE,], 
                  aes(x = logFC, y = -log10(adj.P.Val), 
                      label = SYMBOL), color = "grey", force = 10, max.iter = 5000)+
  geom_hline(yintercept = -log10(0.05), lty = "dotted")+
  theme_classic()
#dev.off()

# Volcano plotting OXPHOS main
#cairo_pdf("deepLRPPRC_Fly/FigureS4F_Phospho_Occupancies_OXPHOS.pdf", 4, 4)
ggplot(data = limma.result, aes(x = logFC, y = -log10(adj.P.Val)))+
  geom_point(data = limma.result[limma.result$mitochondrial != TRUE | 
                                   limma.result$dm.mitocarta$Mitochondrial.Process != 
                                   "Electron Transport Chain",], aes(x = logFC, y = -log10(adj.P.Val)), 
             col = "grey")+
  geom_point(data = limma.result[limma.result$OXPHOS == TRUE,], aes(x = logFC, y = -log10(adj.P.Val)), 
             col = "darkorange3")+
  geom_text_repel(data = limma.result[limma.result$adj.P.Val < 0.05 & 
                                      limma.result$OXPHOS == TRUE,], 
                  aes(x = logFC, y = -log10(adj.P.Val), 
                      label = SYMBOL), color = "darkorange3", force = 10, max.iter = 5000)+
  geom_hline(yintercept = -log10(0.05), lty = "dotted")+
  theme_classic()
#dev.off()

#table.oxphos.phospho <- limma.result[limma.result$dm.mitocarta$Mitochondrial.Process %in% "Electron Transport Chain",]

#write.table(file = "deepLRPPRC_Fly/OXPHOS.phospho.sig.txt", table.oxphos.phospho, row.names = F, 
#            quote = F, sep = "\t")

#boxplot(limma.result$logFC)

limma.result.occ <- limma.result
limma.result.etc <- limma.result.occ[limma.result.occ$dm.mitocarta$Mitochondrial.Process %in% "Electron Transport Chain",]

# Site conservation for OXPHOS
## Prepare a table to manually assess site conservation
phospho.detected <- bsf.phospho.match
phospho.detected$ENSEMBL <- mapIds(org.Dm.eg.db, 
                                   keys=sub(";.*","", bsf.phospho.match$Proteins), 
                                   column="ENSEMBL", 
                                   keytype="UNIPROT",
                                   multiVals="first")

phospho.detected$mitocarta <- dm.mitocarta[match(phospho.detected$ENSEMBL, dm.mitocarta$Gene.ID),]
phospho.detected.oxphos <- phospho.detected[phospho.detected$mitocarta$Mitochondrial.Process %in% "Electron Transport Chain",]

phospho.detected.oxphos.red <- data.frame(name = phospho.detected.oxphos$`Leading proteins`,
                                          site = phospho.detected.oxphos$`Positions within proteins`,
                                          confidence = phospho.detected.oxphos$`Phospho (STY) Probabilities`,
                                          ENSEMBL = phospho.detected.oxphos$mitocarta$Gene.ID,
                                          SYMBOL = phospho.detected.oxphos$mitocarta$SYMBOL,
                                          humanized = phospho.detected.oxphos$mitocarta$humanized,
                                          complex = oxphos.library[match(phospho.detected.oxphos$ENSEMBL, oxphos.library$Dm),]$Complex)

#write.table(phospho.detected.oxphos.red, file = "deepLRPPRC_Fly/OXPHOS.phospho.all.txt", 
 #           row.names = F, quote = F, sep = "\t")

## Manual annotation using DIOPT

phospho.detected_curated <- read.table(file = "deepLRPPRC_Fly/OXPHOS.phospho.all_curated.txt", sep = "\t", header = T)

phospho.detected_curated$occ.L <- Occupancy.L[match(phospho.detected_curated$short, rownames(Occupancy.L)),2:5]
phospho.detected_curated$occ.H <- Occupancy.H[match(phospho.detected_curated$short, rownames(Occupancy.H)),2:5]
phospho.detected_curated$mean <- Occupancy[match(phospho.detected_curated$short, rownames(Occupancy)),]

phospho.detected_curated_w.occ <- write.table(phospho.detected_curated, file = "deepLRPPRC_Fly/OXPHOS.phospho.all_curated_w.occ.txt", sep = "\t")


phospho.detected_curated$mutated.from <- sub("\\..*","",phospho.detected_curated$AA.Dm)
phospho.detected_curated$mutated.to <- sub("\\..*","",phospho.detected_curated$AA.Hs)
phospho.detected_curated$combo <- paste(phospho.detected_curated$mutated.from, ">", phospho.detected_curated$mutated.to, sep = "")

phospho.alluvial <- data.frame(table(phospho.detected_curated$combo))
phospho.alluvial$from <- sub(">.*", "", phospho.alluvial$Var1)
phospho.alluvial$to <- sub(".*>", "", phospho.alluvial$Var1)

is_alluvia_form(as.data.frame(phospho.alluvial), axes = 2:4, silent = TRUE)

#cairo_pdf("deepLRPPRC_Fly/Figure4B_OXPHOS.sites.mutate.to.pdf")
ggplot(as.data.frame(phospho.alluvial),
       aes(y = Freq, axis1 = from, axis2 = to))+
  scale_fill_brewer(type = "qual", palette = "Set2") +
  geom_flow(aes(fill = from), stat = "alluvium", lode.guidance = "rightleft",
            color = "darkgray") +
  geom_stratum(width = 1/12, fill = "white", color = "black") +
  geom_label(stat = "stratum", label.strata = TRUE) +
  guides(fill = FALSE)
#dev.off()

## Site conservation by complex
phospho.by.complex <- data.frame(complex = phospho.detected_curated$complex,
                                 conserved = phospho.detected_curated$conserved)

phospho.by.complex$combo <- paste(phospho.by.complex$complex, phospho.by.complex$conserved, sep =".")
phospho.by.complex <- data.frame(table(phospho.by.complex$combo))
phospho.by.complex$complex <- sub("\\..*","",phospho.by.complex$Var1)
phospho.by.complex$conserved <- sub(".*\\.","",phospho.by.complex$Var1)

#cairo_pdf("deepLRPPRC_Fly/FigureS4G_OXPHOS.sites.conserved.by.complex.pdf", 6, 4)
ggplot(data = phospho.by.complex, aes(x = complex, y = Freq, fill = conserved))+
  geom_bar(stat="identity", position=position_dodge())+
  scale_fill_brewer(palette="Dark2")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
#dev.off()

## Are processes under-/overrepresented in mitocarta in comparison to the peptides detected?
phospho.detected$mitocarta <- dm.mitocarta[match(phospho.detected$ENSEMBL, dm.mitocarta$Gene.ID),]
table(phospho.detected$mitocarta$Mitochondrial.Process)

mod.spec.pept.lrpprc <- read_tsv("deepLRPPRC_Fly/IA_FS_LRPPRC_SILAF_Phospho_with_match_06_2019/data/modificationSpecificPeptides.txt") # read in peptide files

mod.spec.pept.lrpprc$ENSEMBL <- mapIds(org.Dm.eg.db, 
                                       keys=sub(";.*","", mod.spec.pept.lrpprc$Proteins), 
                                       column="ENSEMBL", 
                                       keytype="UNIPROT",
                                       multiVals="first")

mod.spec.pept.lrpprc$mitocarta <- dm.mitocarta[match(mod.spec.pept.lrpprc$ENSEMBL, dm.mitocarta$Gene.ID),]

### Prepare a table showing ratios
mod.spec.pept.lrpprc.unmod <- mod.spec.pept.lrpprc[mod.spec.pept.lrpprc$Modifications == "Unmodified",]

relative.ratios <- data.frame(a.proteins = as.vector(table(dm.mitocarta$Mitochondrial.Process)),
                              b.peptides = as.vector(table(mod.spec.pept.lrpprc.unmod$mitocarta$Mitochondrial.Process)),
                              c.phospho = as.vector(table(phospho.detected$mitocarta$Mitochondrial.Process)))

rownames(relative.ratios) <- names(table(mod.spec.pept.lrpprc$mitocarta$Mitochondrial.Process))

col.sumns <- colSums(relative.ratios)
relative.ratios <- data.frame(t(t(relative.ratios)/col.sumns))
relative.ratios$cats <- rownames(relative.ratios)

relative.ratios <- gather(relative.ratios, "level", "observed", 1:3)

#cairo_pdf(file = "deepLRPPRC_Fly/Figure4A_Overrepresented.per.category.pdf", 10, 3)
ggplot(data=relative.ratios, aes(x=cats, y=observed, fill=level)) +
  geom_bar(stat="identity", position=position_dodge())+
  scale_fill_brewer(palette="Dark2")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
#dev.off()

## Comparing mito enrichment with abundance of phospho-sites

### Proteins
total <- total.proteome.after.enrich
total <- total[!is.na(total$logFC),]
total <- total[!is.na(total$ENSEMBL),]

table(total$mitochondrial)
dim(dm.mitocarta)

### Peptides unmodified
table(is.na(mod.spec.pept.lrpprc.unmod$mitocarta$Gene.ID))

### Peptides phosphorylated
mod.spec.pept.lrpprc.phospho <- mod.spec.pept.lrpprc[grepl("Phospho.*", mod.spec.pept.lrpprc$Modifications),]
table(is.na(mod.spec.pept.lrpprc.phospho$mitocarta$Gene.ID))

### Phospho-sites
table(is.na(phospho.detected$mitocarta$Gene.ID))

### Prepare a table
mito.enrichment <- data.frame(a.mitocarta = as.vector(table(-(dm.mitocarta$Gene.ID %in% total$ENSEMBL))),
                              b.proteome = as.vector(table(-total$mitochondrial)),
                              c.proteome.phospho = as.vector(table(is.na(phospho.detected$mitocarta$Gene.ID))),
                              d.peptides.unmod = as.vector(table(is.na(mod.spec.pept.lrpprc.unmod$mitocarta$Gene.ID))),
                              e.peptides.phospho = as.vector(table(is.na(mod.spec.pept.lrpprc.phospho$mitocarta$Gene.ID))))

rownames(mito.enrichment) <- c("mito", "not.mito")

col.sums <- colSums(mito.enrichment)
ratio.mito.enrichment <- data.frame(t(t(mito.enrichment)/col.sums))

ratio.mito.enrichment$compartment <- c("b.mito", "a.not.mito")
ratio.mito.enrichment <- gather(ratio.mito.enrichment, "type", "ratios", 1:5)

#cairo_pdf(file = "deepLRPPRC_Fly/FigureS4C_Mito.enrichment.pdf", 4, 3)
ggplot(data=ratio.mito.enrichment, aes(x=type, y=ratios, fill=compartment)) +
  geom_bar(stat="identity")+
  scale_fill_brewer(palette="Dark2")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
#dev.off()

#R code used for characterising specific targets will be released upon publication or upon reviewer demand. 
```